keymanapp/keyman
# fix(android/engine): Use fixed sizes for preview keys

Fixes #6232 where the preview keys in landscape orientation were getting truncated.

Instead of scaling the preview key as the OSK height is changed, this uses fixed sizes for preview keys. This way the keycap doesn't get truncated.

Dimensions for tablets not affected because we don't display preview keys for tablet. (Longpress / subkeys are displayed though).

Note: this PR does not address the base key issue when the adjusted keyboard height is set to minimums (noted in a comment on #6232). I'll save that for future PRs.

Screenshot of default landscape OSK
![g landscape](https://user-images.githubusercontent.com/7358010/160974291-e3a75df4-a8b8-47df-954d-21f66499c2f0.png)

## User Testing
For these tests, "short-press on a key" refers to typing on Keyman keyboard either in-app or system keyboard (e.g. in a 3rd app like Chrome). Depends on the test group

**Setup**
1. On an Android device/emulator, install the PR build of "Keyman for Android"

GROUP_INAPP_KEYBOARD: Test preview keys in in-app keyboard
GROUP_SYSTEM_KEYBOARD: Test preview keys in Keyman system keyboard

* **TEST_KEY_PREVIEW**
1. With the device in *Portrait* orientation, launch "Keyman for Android
2.  Dismiss the "Getting Started" menu if visible
3. Short-press on several keys on the default layer and observe the key preview (before the long-press keys display). Offending keys on the issue include: `q`, `y`, `p`, `g`.
4. Verify the entire key previews are visible and not truncated.
5. From the Keyman app --> Settings --> Adjust keyboard height
6. On the "Adjust keyboard height" menu, drag the OSK all the way to the minimum height for Portrait orientation
7. Hit back keys to exit the menus
8. Short-press on several keys on the default layer and observe the key preview
9. Verify the entire key previews are visible and not truncated.
10. Rotate the device to *Landscape* orientation
11. Short-press on several keys on the default layer and observe the key preview
12. Verify the entire key previews are visible and not truncated.
13. From the Keyman app --> Settings --> Adjust keyboard height
14. On the "Adjust keyboard height" menu, drag the OSK all the way to the minimum height for Landscape orientation
15. Hit back keys to exit the menus
16. Short-press on several keys on the default layer and observe the key preview
17. Verify the entire key previews are visible and not truncated.
 
Screenshot of landscape OSK at minimum height
![q landscape - minimum](https://user-images.githubusercontent.com/7358010/160973998-db5877f4-9c9d-422b-8856-a8b985982850.png)
